src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs

```diff
@@ -316,18 +332,30 @@ public static int Serialize(byte[] buffer, int cursor, object obj)
                 return SerializeMap(buffer, cursor, v);
             case object[] v:
                 return SerializeArray(buffer, cursor, v);
+
 #if NET6_0_OR_GREATER
             case ISpanFormattable v:
                 tmp = stackalloc char[MAX_STACK_ALLOC_SIZE_IN_BYTES / sizeof(char)];
-                if (v.TryFormat(tmp, out charsWritten, default, CultureInfo.InvariantCulture))
+                if (v.TryFormat(tmp, out charsWritten, default, CultureInfo.InvariantCulture) && charsWritten > 0)
```

Comment:
Seems a bit odd to have the `&& charsWritten > 0` here and the `value.IsEmpty` check above. Do we need both? Is there a particular test case you have in mind for this?

Reply:
Right. The `value` check is not really needed as we already do a validation at the call site. I just left it in there for now as I thought I might reuse that function later on. I could remove that check and add a `Debug.Assert` there, if you'd like.

Reply:
I think switching to `Debug.Assert(!value.IsEmpty, "value was empty")` inside `SerializeString` would be fine. I'm not sure if we need the `&& charsWritten > 0` either TBH. If `TryFormat` returns `true` I think we trust that something was formatted. Up to you 😄